### PR TITLE
extendObservable can be used late and support non-enumerable getters.

### PR DESCRIPTION
**Pull request by MoonBall**
### Code change checklist

-   [x] Added/updated unit tests
-   [ ] Updated changelog
-   [ ] Updated `/docs`. For new functionality, at least `API.md` should be updated
-   [x] Verified that there is no significant performance drop (`npm run perf`)

Fix https://github.com/mobxjs/mobx/issues/2386 on v5.

<a href="https://gitpod.io/#https://github.com/mobxjs/mobx/pull/2412"></a>

### CODE REVIEW

**Comment by mweststrate** [comment]
Looking great! As far as I'm concerned `enableNonEnumerablePropertiesInExtendObservable` could be just the default and I don't think it has to be an option. Afaik all normal use cases of extendObservable consist of the second argument to be a locally constructed object literal (and not a class instance or something), so I expect this change to be safe.

**Comment by MoonBall** [comment]
> Looking great! As far as I'm concerned `enableNonEnumerablePropertiesInExtendObservable` could be just the default and I don't think it has to be an option. Afaik all normal use cases of extendObservable consist of the second argument to be a locally constructed object literal (and not a class instance or something), so I expect this change to be safe.

okey, I remove the `enableNonEnumerablePropertiesInExtendObservable` state. PTAL.

**Comment by mweststrate** [comment]
Awesome, thanks for doing this! Would you mind making the same change in the v4 folder? (from 6 onward that wouldn't be needed anymore luckily)

File: src/v4/api/extendobservable.ts [hunk 1]
```diff
@@ -64,7 +64,7 @@ export function extendObservable<A extends Object, B extends Object>(
     asObservableObject(target, options.name, defaultDecorator.enhancer) // make sure object is observable, even without initial props
     startBatch()
     try {
-        for (let key in properties) {
```
Comment (MoonBall) [inline src/v4/api/extendobservable.ts:67]: `for ... in ...` contains enumerable properties that the `properties` inherits from its prototype chain. 
But `Object.getOwnPropertyNames` only contains enumerable properties of `properties` itself. 
@mweststrate I'm not sure the difference matters.

Comment (mweststrate) [reply src/v4/api/extendobservable.ts:67]: @MoonBall good one, no for extendObservable that should be match. However, If I remember correctly the downlevel compilation of `for of` loops is problematic with MobX 4, which might run in environments without iterators. Would you mind changing it to a `.forEach` or class `for(;;)` loop?

Comment (MoonBall) [reply src/v4/api/extendobservable.ts:67]: I fix `for of` by using `for(;;)`.

**Comment by MoonBall** [comment]
>  If you want to contribute to MobX more in the future, let me know and I send a maintainers invite. 

Thanks, I'm interested in doing it.

**Pull request merged**